File: SevenNightClassPCS/oop-java/tictactoe1/TicTacToe.java

public class TicTacToe {
	
	public static void main(String [] args) {
		
		Board brd = new Board();
		HumanPlayer player1 = new HumanPlayer(brd,'X');
		MrRandom player2 = new MrRandom(brd,'O');
		String status = "";
		
		while(true) {
			int move = player1.getMove();
			brd.makeMove(move, 'X');
			
			if(brd.getStatus()=='X') {
				status = "Player X wins!";
				break;
			}
			
			if(brd.getStatus()=='C') {
				status = "It is a Tie!";
				break;
			}
			
			move = player2.getMove();
			brd.makeMove(move, 'O');
			
			if(brd.getStatus()=='O') {
				status = "Player O wins!";
				break;
			}
		}
		
		System.out.println(brd);
		System.out.println(status);
		
	}

}

File: Topics/01_Encapsulation/01_05_TicTacToe_example/cpp_source/TicTacToe.cpp
#include <iostream>
using namespace std;

#include "Board.h"
#include "MrRandom.h"
#include "HumanPlayer.h"

int main(int argc, char ** argv) {

	Board * brd = new Board();
	string r = brd->getStringRepr();
	cout << r << endl;


	delete brd;

}
